### DISCUSSION TITLE
How do you search through the code to understand things further?

### DISCUSSION THREAD

**Discussion by visualcode-t** [bevyengine/bevy#1067] [Q&A] [answered]
I am new to GitHub, and I'm really struggling with figuring out how some things in Bevy work. I was using Crates.io, but a lot of the information I'm finding here isn't compatible with Crates.io. So, I switched to using GitHub in my Cargo.toml in order to reduce some confusion. However, now, as an example, I'm playing with the ButtonBundle, and I'm not sure of all its parameters. All my searches for "ButtonBundle" come up empty, though , and docs.rs (which I had been using) seems to be on the Crates.io version.

I am using Visual Studio Code and I have tried using that to follow the source, but it doesn't seem to be able to do that with Bevy.

So what is the proper way to dive a little deeper into something when I am not sure how it works?

**Comment by cart** [comment, marked as answer]
I recommend installing the Rust-Analyzer vscode plugin: https://github.com/rust-analyzer/rust-analyzer

Then you should be able to do things like:

F12: navigate to symbol source
Ctrl-o: search symbols in current file
Ctrl-shift-o: search all symbols in the project

**Reply by visualcode-t** [reply]
Wow, that's a HUGE improvement! Thank you for that.

**Comment by mockersf** [comment]
you can also clone this repo and build the documentation locally.

I usually build it like this:
`cargo doc --no-deps --all-features --open`

**Reply by visualcode-t** [reply]
Also good to know, I'll get a local copy going. Thank you.
